Automattic/pocket-casts-ios
# Downloads management: Implement banner on Downloads screen

| 📘 Part of: #2430  |  
|:---:|

Fixes #2431  

| 1 | 2 | 3 |
| - | - | - |
| ![Simulator Screenshot - iPhone 16 Pro - 2024-11-18 at 11 27 39](https://github.com/user-attachments/assets/d778e10f-e763-4c9c-8a2b-8741767bbf3a) | ![Simulator Screenshot - iPhone 16 Pro - 2024-11-18 at 11 27 26](https://github.com/user-attachments/assets/7aed8f76-5a7a-48e3-a3d0-e555e7afaffe) | ![Simulator Screenshot - iPhone 16 Pro - 2024-11-18 at 11 26 41](https://github.com/user-attachments/assets/9c2e5e3e-0af9-49ac-baa3-e9eba188b8bb) |


This PR adds the download management banner on the downloads screen

## To test

To make it easier to see the banner change the `percentage` value to `1 ` in `shouldShowBanner` method in `ManageDownloadsBannerView.swift` file. 

1. Start the app
2. Ensure that you have the `manageDownloadedEpisodes` FF active and the `tracksLogging` FF
3. Download some episode files
4. Go to Profile -> Downloads
5. Check if you see the banner on the top and the event `freeUpSpaceBannerShown` on the console
6. Tap on `Manage Downloads`
7. Check if the files screen is show.
8. Check if the event: `freeUpSpaceManageDownloadsTapped` is show in console with the source: `downloads` property


## Checklist
